Example doesn't work

I am getting the following error:

```
error[E0277]: the trait bound `futures::future::FutureResult<thruster::BasicContext, _>: futures::future::Future` is not satisfied
  --> src/main.rs:21:3
   |
21 |   Box::new(future::ok(context))
   |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `futures::future::Future` is not implemented for `futures::future::FutureResult<thruster::BasicContext, _>`
   |
   = note: required for the cast to the object type `futures::future::Future<Item=thruster::BasicContext, Error=std::io::Error> + std::marker::Send`
```

Also, it seems that there is no need to list `serde`, `serde_json`, and `tokio` explicitly in the example project.

Rust version: 1.27 and Nightly (2018-06-19).

P.S. I had to comment `#![feature(test)]` in the `lib.rs` to get it compiling to this state on stable Rust.

--- a/README.md
+++ b/README.md
@@ -51,19 +51,17 @@ Based on frameworks like Koa, and Express, thruster aims to be a pleasure to dev
 ```rust
 extern crate thruster;
 extern crate futures;
-extern crate serde;
-extern crate serde_json;
-extern crate tokio;
 
 use std::boxed::Box;
 use futures::future;
 
 use thruster::{App, BasicContext as Ctx, MiddlewareChain, MiddlewareReturnValue, Request};
 
-fn generate_context(request: &Request) -> Ctx {
+fn generate_context(request: Request) -> Ctx {
   Ctx {
     body: "".to_owned(),
-    params: request.params().clone()
+    params: request.params,
+    query_params: request.query_params
   }
 }
 
@@ -79,7 +77,7 @@ fn main() {
 
   let mut app = App::<Ctx>::create(generate_context);
 
-  app.get("/plaintext", smallvec![plaintext]);
+  app.get("/plaintext", vec![plaintext]);
 
   App::start(app, "0.0.0.0", 4321);
 }

--- a/examples/most_basic.rs
+++ b/examples/most_basic.rs
@@ -1,10 +1,5 @@
-#![feature(test)]
-
 extern crate thruster;
 extern crate futures;
-extern crate serde;
-extern crate serde_json;
-extern crate tokio;
 
 use std::boxed::Box;
 use futures::future;
@@ -14,8 +9,8 @@ use thruster::{App, BasicContext as Ctx, MiddlewareChain, MiddlewareReturnValue,
 fn generate_context(request: Request) -> Ctx {
   Ctx {
     body: "".to_owned(),
-    params: request.params().clone(),
-    query_params: request.query_params().clone()
+    params: request.params,
+    query_params: request.query_params
   }
 }
 
@@ -31,7 +26,7 @@ fn main() {
 
   let mut app = App::<Ctx>::create(generate_context);
 
-  app.get("/plaintext", smallvec![plaintext]);
+  app.get("/plaintext", vec![plaintext]);
 
   App::start(app, "0.0.0.0", 4321);
 }

--- a/src/app.rs
+++ b/src/app.rs
@@ -252,7 +252,8 @@ impl<T: Context + Send> App<T> {
 
 #[cfg(test)]
 mod tests {
-  use test::Bencher;
+  // Uncomment to run benchmarks
+  // use test::Bencher;
   use super::*;
   use std::collections::HashMap;
   use bytes::{BytesMut, BufMut};
@@ -296,71 +297,72 @@ mod tests {
     }
   }
 
-  #[bench]
-  fn bench_route_match(bench: &mut Bencher) {
-    let mut app = App::<BasicContext>::new();
+  // Uncomment to run benchmarks
+  // #[bench]
+  // fn bench_route_match(bench: &mut Bencher) {
+  //   let mut app = App::<BasicContext>::new();
 
-    fn test_fn_1(context: BasicContext, _chain: &MiddlewareChain<BasicContext>) -> Box<Future<Item=BasicContext, Error=io::Error> + Send> {
-      Box::new(future::ok(BasicContext {
-        body: "world".to_owned(),
-        params: HashMap::new(),
-        query_params: context.query_params
-      }))
-    };
+  //   fn test_fn_1(context: BasicContext, _chain: &MiddlewareChain<BasicContext>) -> Box<Future<Item=BasicContext, Error=io::Error> + Send> {
+  //     Box::new(future::ok(BasicContext {
+  //       body: "world".to_owned(),
+  //       params: HashMap::new(),
+  //       query_params: context.query_params
+  //     }))
+  //   };
 
-    app.get("/test/hello", vec![test_fn_1]);
+  //   app.get("/test/hello", vec![test_fn_1]);
 
-    bench.iter(|| {
-      let mut bytes = BytesMut::with_capacity(47);
-      bytes.put(&b"GET /test/hello HTTP/1.1\nHost: localhost:8080\n\n"[..]);
-      let request = decode(&mut bytes).unwrap().unwrap();
-      let _response = app.resolve(request).wait().unwrap();
-    });
-  }
+  //   bench.iter(|| {
+  //     let mut bytes = BytesMut::with_capacity(47);
+  //     bytes.put(&b"GET /test/hello HTTP/1.1\nHost: localhost:8080\n\n"[..]);
+  //     let request = decode(&mut bytes).unwrap().unwrap();
+  //     let _response = app.resolve(request).wait().unwrap();
+  //   });
+  // }
 
-  #[bench]
-  fn bench_route_match_with_param(bench: &mut Bencher) {
-    let mut app = App::<BasicContext>::new();
+  // #[bench]
+  // fn bench_route_match_with_param(bench: &mut Bencher) {
+  //   let mut app = App::<BasicContext>::new();
 
-    fn test_fn_1(context: BasicContext, _chain: &MiddlewareChain<BasicContext>) -> Box<Future<Item=BasicContext, Error=io::Error> + Send> {
-      Box::new(future::ok(BasicContext {
-        body: context.params.get("hello").unwrap().to_owned(),
-        params: HashMap::new(),
-        query_params: context.query_params
-      }))
-    };
+  //   fn test_fn_1(context: BasicContext, _chain: &MiddlewareChain<BasicContext>) -> Box<Future<Item=BasicContext, Error=io::Error> + Send> {
+  //     Box::new(future::ok(BasicContext {
+  //       body: context.params.get("hello").unwrap().to_owned(),
+  //       params: HashMap::new(),
+  //       query_params: context.query_params
+  //     }))
+  //   };
 
-    app.get("/test/:hello", vec![test_fn_1]);
+  //   app.get("/test/:hello", vec![test_fn_1]);
 
-    bench.iter(|| {
-      let mut bytes = BytesMut::with_capacity(48);
-      bytes.put(&b"GET /test/world HTTP/1.1\nHost: localhost:8080\n\n"[..]);
-      let request = decode(&mut bytes).unwrap().unwrap();
-      let _response = app.resolve(request).wait().unwrap();
-    });
-  }
+  //   bench.iter(|| {
+  //     let mut bytes = BytesMut::with_capacity(48);
+  //     bytes.put(&b"GET /test/world HTTP/1.1\nHost: localhost:8080\n\n"[..]);
+  //     let request = decode(&mut bytes).unwrap().unwrap();
+  //     let _response = app.resolve(request).wait().unwrap();
+  //   });
+  // }
 
-  #[bench]
-  fn bench_route_match_with_query_param(bench: &mut Bencher) {
-    let mut app = App::<BasicContext>::new();
+  // #[bench]
+  // fn bench_route_match_with_query_param(bench: &mut Bencher) {
+  //   let mut app = App::<BasicContext>::new();
 
-    fn test_fn_1(context: BasicContext, _chain: &MiddlewareChain<BasicContext>) -> Box<Future<Item=BasicContext, Error=io::Error> + Send> {
-      Box::new(future::ok(BasicContext {
-        body: context.query_params.get("hello").unwrap().to_owned(),
-        params: HashMap::new(),
-        query_params: context.query_params
-      }))
-    };
+  //   fn test_fn_1(context: BasicContext, _chain: &MiddlewareChain<BasicContext>) -> Box<Future<Item=BasicContext, Error=io::Error> + Send> {
+  //     Box::new(future::ok(BasicContext {
+  //       body: context.query_params.get("hello").unwrap().to_owned(),
+  //       params: HashMap::new(),
+  //       query_params: context.query_params
+  //     }))
+  //   };
 
-    app.get("/test", vec![test_fn_1]);
+  //   app.get("/test", vec![test_fn_1]);
 
-    bench.iter(|| {
-      let mut bytes = BytesMut::with_capacity(54);
-      bytes.put(&b"GET /test?hello=world HTTP/1.1\nHost: localhost:8080\n\n"[..]);
-      let request = decode(&mut bytes).unwrap().unwrap();
-      let _response = app.resolve(request).wait().unwrap();
-    });
-  }
+  //   bench.iter(|| {
+  //     let mut bytes = BytesMut::with_capacity(54);
+  //     bytes.put(&b"GET /test?hello=world HTTP/1.1\nHost: localhost:8080\n\n"[..]);
+  //     let request = decode(&mut bytes).unwrap().unwrap();
+  //     let _response = app.resolve(request).wait().unwrap();
+  //   });
+  // }
 
   #[test]
   fn it_should_execute_all_middlware_with_a_given_request() {

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -1,4 +1,5 @@
-#![feature(test)]
+// Uncomment to run benchmarks
+// #![feature(test)]
 
 extern crate bytes;
 extern crate futures;
@@ -21,8 +22,10 @@ extern crate tokio_io;
 #[macro_use] extern crate serde_json;
 #[allow(unused_imports)]
 extern crate tokio_core;
-#[allow(unused_imports)]
-extern crate test;
+
+// Uncomment to run benchmarks
+// #[allow(unused_imports)]
+// extern crate test;
 
 mod app;
 mod builtins;

--- a/src/route_parser.rs
+++ b/src/route_parser.rs
@@ -122,7 +122,6 @@ mod tests {
 
     let matched = route_parser.match_route("1/2/3");
     assert!(matched.middleware.len() == 1);
-    // assert!(matched.middleware.get(0).unwrap() == &(test_function as Middleware));
   }
 
   #[test]

